src/vm/hints/set.rs

```diff
@@ -341,7 +343,7 @@ mod tests {
         assert_eq!(
             execute_hint(&mut vm, HINT_CODE, ids, &ApTracking::new()),
             Err(VirtualMachineError::ExpectedInteger(
-                MaybeRelocatable::from((7, 8))
+                MaybeRelocatable::from((0, 3))
```

Comment:
Error is now thrown by get_elem_from_var_name, instead of being created inside the hint. The first them used the address of the variable and the other one used the variable's current value. The value was checked so that the logic of the test remains unchanged